src/internal/components/selectable-item/styles.scss

```diff
@@ -124,6 +125,52 @@
     }
   }
 
+  &.sticky {
+    position: sticky;
+    inset-block-start: 0;
+
+    // Push the next option down to prevent its border in highlighted or selected state from being partially covered by the sticky option
+    margin-block-end: calc(#{awsui.$border-item-width} - #{awsui.$border-divider-list-width});
+
+    // Stay on top of the other options when they are scrolled up
+    z-index: 4;
+
+    &:not(.highlighted):not(.selected) {
+      // Prevent covering the list border despite the higher z-index
+      border-inline-start-width: awsui.$border-item-width;
+      border-inline-start-color: awsui.$color-border-dropdown-container;
+      border-inline-end-color: awsui.$color-border-dropdown-container;
+      padding-inline: styles.$control-padding-horizontal;
+
+      &:not(.with-scrollbar) {
```

Comment:
When a scrollbar is present in the list menu, we need to remove the right border and the top-right border radius in order to prevent this:

<img width="78" alt="Screenshot 2025-03-19 at 16 38 12" src="https://github.com/user-attachments/assets/4b87d019-652b-4cb8-8d9f-ddfc2c1acd7f" />